src/parser/type_checking.c

```diff
@@ -5446,6 +5446,8 @@ pt_coerce_range_expr_arguments (PARSER_CONTEXT * parser, PT_NODE * expr, PT_NODE
 	{
 	  PT_NODE *temp = NULL;
 	  int precision = 0, scale = 0;
+	  int units = data_type->info.data_type.units; /* code set */
+	  int collation_id = data_type->info.data_type.collation_id; /* collation_id */
```

Comment:
`data_type` was newly created from line 5437. So these values may be initialized with a garbage value.
```suggestion
	  int units = LANG_SYS_CODESET; /* code set */
	  int collation_id = LANG_SYS_COLLATION; /* collation_id */
```

Reply:
I agree.